src/plugins/kibana/public/settings/sections/kibi_relations/controllers/relations.js

```diff
@@ -26,7 +26,7 @@ define(function (require) {
             scope.$apply(function () {
               ngModelCtrl.$setViewValue(elm.val());
             });
-          }, attr.ngDebounce || 1000);
+          }, attr.ngDebounce || 1);
```

Comment:
Hi, what you call a hack is what we actually wanted to do for some time which is to disable the animation completely. They are nice when you have just a few relations but become a problem when you have many. And your changes are doing exactly that.

I've tested it today and did not encounter a problem with relations become "red".
I'll give it another shot tomorrow (but it could be that the problem was fixed already on 4.6.4 branch on which I'm testing this, however, I've tested with my change about kibiDebounce)

About the kibi-debounce - angular converts the attribute names to camel case so 
**kibi-debounce** become **attr.kibiDebounce** inside the directive.